include/vapor/GeoImageTMS.h

```diff
@@ -9,13 +9,78 @@
 #include <xtiffio.h>
 #include <geotiff.h>
 #endif
+#include <sstream>
+#include <fstream>
+#include <sys/stat.h>
 #include <vapor/MyBase.h>
 #include <vapor/UDUnitsClass.h>
 #include "GeoTileMercator.h"
 #include "GeoImage.h"
 
 namespace VAPoR {
 
+static inline bool IsTMSFile( 
+    std::string path 
+) {
+    if ( path.rfind(".tms", path.size()-4) != string::npos ) {
+        ifstream in;
+        in.open( path.c_str() );
+        if ( ! in ) {
+            GeoImage::SetErrMsg("fopen(%s) : %M", path.c_str());
+            return false ;
+        }
+        in.close();
+    }
+
+    else {
+        return false;
+    }
+
+    return true;
+}
+
+// Get the file path of a single tile from the TMS database
+// with the give lod and tile coordinates
+//
+static inline std::string TilePath(
```

Comment:
We don't have a coding convention about this, but typically functions that are more than a couple of lines of code are defined in .cpp files, not .h files. Is there are reason for defining this function in the header file?

Reply:
> Is there are reason for defining this function in the header file?

Yes, TilePath() is a helper function used in the render and vaporgui libraries.  GeoImageTMS.cpp and ImageSubtabs.cpp are sharing the logic behind this function, by including GeoImageTMS.h.

GeoImageTMS.cpp is making use of TilePath() directly.

ImageSubtabs.cpp calls TilePath() indirectly by calling GetAvailableTMSLODs(), which is also found in GeoImageTMS.h.

Reply:
I'm a little confused. I don't see how these functions are any different from any other public static functions, and why they cannot simply be *declared* in the .h file and *defined* in the .cpp file

Reply:
Sorry - I misunderstood the question.

When we met last week I thought we planned on making these helper methods static, and they would be declared outside of the GeoImageTMS class.  They were inlined so they could be externally linked.

But if they're static public member functions, they don't need to be inlined, so I've made that change.

Reply:
Whether they are static member functions or simply free functions they can still be made visible to other modules by simply declaring them in the .h files and defining them in a .cpp file that is linked as a library.

Reply:
True, however I'm pretty sure that static free functions must be inlined.  But I understand this is not what we wanted.